x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/plan/logical/Enrich.java

```diff
@@ -284,6 +286,25 @@ private void checkForPlansForbiddenBeforeRemoteEnrich(Failures failures) {
         fails.forEach(f -> failures.add(fail(this, "ENRICH with remote policy can't be executed after [" + f.text() + "]" + f.source())));
     }
 
+    private void checkMvExpandAfterLimit(Failures failures) {
+        this.forEachDown(MvExpand.class, u -> {
+            u.forEachDown(p -> {
+                if (p instanceof Limit || p instanceof TopN) {
+                    failures.add(fail(this, "MV_EXPAND after LIMIT is incompatible with remote ENRICH"));
```

Comment:
note: technically, that's only true if we cannot push the remote enrich past the mv_expand. Which we sometimes could! (There are more optimizations that could be applied to `MV_EXPAND`, in general.)

But since we currently don't do this, this check will strictly prohibit only queries that we can't properly run anyway, so this is fine.

Maybe we could add a comment, though?